src/botkit-middleware-witai.js

```diff
@@ -29,9 +29,9 @@ module.exports = function(config) {
     var middleware = {};
 
     middleware.receive = function(bot, message, next) {
-        // Only parse messages of type text and mention the bot.
+        // Only parse messages of type text and mention the bot (or when the bot is a web based bot where it's the only thing to talk to).
         // Otherwise it would send every single message to wit (probably don't want that).
-        if (message.text && message.text.indexOf(bot.identity.id) > -1) {
+        if (message.text && (bot._controller._config.adapter.name === 'Web Adapter' || message.text.indexOf(bot.identity.id) > -1)) {
```

Comment:
this will break compatibility with botkit 0.7 because bot._controller is not defined.

but message.text.indexOf(bot.identity.id) is a terrible way of testing whether the bot is being mentioned.

i think we should just remove that part, and do a test on if message.text

Reply:
Good point, submitted an edited version